vertx-web/src/main/java/io/vertx/ext/web/handler/impl/ErrorHandlerImpl.java

```diff
@@ -56,50 +64,72 @@ public void handle(RoutingContext context) {
     HttpServerRequest request = context.request();
     HttpServerResponse response = context.response();
 
-    if (context.statusCode() != -1) {
-      response.setStatusCode(context.statusCode());
-    } else {
-      // Internal error
-      response.setStatusCode(500);
+    Throwable failure = context.failure();
+    
+    if(answerWithKnownFailure(context)){
+      return;
     }
-
-    // does the response already set the mime type?
-    String mime = response.headers().get(HttpHeaders.CONTENT_TYPE);
-
-    int errorCode;
+    
+    int errorCode = context.statusCode();
     String errorMessage = null;
-    if (context.statusCode() != -1) {
-      errorCode = context.statusCode();
+    if (errorCode != -1) {
+      context.response().setStatusCode(errorCode);
       errorMessage = context.response().getStatusMessage();
     } else {
       errorCode = 500;
       if (displayExceptionDetails) {
-        errorMessage = context.failure().getMessage();
+        errorMessage = failure.getMessage();
       }
       if (errorMessage == null) {
         errorMessage = "Internal Server Error";
       }
       // no new lines are allowed in the status message
       response.setStatusMessage(errorMessage.replaceAll("\\r|\\n", " "));
     }
-
-    if (mime != null) {
-      if (sendError(context, mime, errorCode, errorMessage)) {
-        return;
-      }
+    
+    answerWithError(context, errorCode, errorMessage);
+  }
+  
+  private boolean answerWithKnownFailure(RoutingContext context) {
+    Throwable failure = context.failure();
+    
+    if(failure instanceof HeaderTooLongException){
+      // Math.random is random enough for this purpose. Will wait between 1 ms and 11ms
+      answerDelayTimer.schedule(
+          new DelayedAnswer(context, 400, "A header was too long to process"), 1 + (long)(Math.random() * 10)
```

Comment:
Is it a good idea to rely on a random value? it introduces complexity and could for example introduce false failures if the server is under heavy load/gc'ing etc...